src/http/ngx_http_wasm_api.c

```diff
@@ -359,7 +359,128 @@ int32_t
 proxy_set_property(int32_t path_data, int32_t path_size,
                    int32_t data, int32_t size)
 {
-    return PROXY_RESULT_OK;
+    u_char                      *p;
+    u_char                      *key_lowcase;
+    ngx_log_t                   *log;
+    ngx_uint_t                   hash;
+    const u_char                *key;
+    const u_char                *value;
+    ngx_http_request_t          *r;
+    ngx_http_variable_t         *v;
+    ngx_http_variable_value_t   *vv;
+    ngx_http_core_main_conf_t   *cmcf;
+
+    log = ngx_http_wasm_get_log();
+    must_get_req(r);
+
+    /* fetch property key and value */
+    key = ngx_wasm_vm.get_memory(log, path_data, path_size);
+    if (key == NULL) {
+        return PROXY_RESULT_INVALID_MEMORY_ACCESS;
+    }
+    value = ngx_wasm_vm.get_memory(log, data, size);
+    if (key == NULL) {
+        return PROXY_RESULT_INVALID_MEMORY_ACCESS;
+    }
+
+    /*
+     * Request a piece of temporary memory to store the
+     * lowercase characters of the property key.
+     */
+    key_lowcase = ngx_http_wasm_get_string_buf(r->pool, path_size);
+    hash = ngx_hash_strlow(key_lowcase, (u_char *) key, path_size);
+
+    /*
+     * Get the ngx_http_core_module main configuration block
+     * and search variables from it.
+     */
+    cmcf = ngx_http_get_module_main_conf(r, ngx_http_core_module);
```

Comment:
We can save lots of code by calling ngx_http_lua_ffi_var_set directly? See https://github.com/openresty/lua-resty-core/blob/master/lib/resty/core/var.lua#L134

Reply:
Yes, direct reuse is the best way, I will try it out.

Reply:
changed